### PR TITLE
Adds tippable buckets and fireaxes (as a proof of concept)

### PR DESCRIPTION
**Pull request by optimumtact**
## About The Pull Request
Lets you prop things on top of airlocks that fall on people when they open them
## Why It's Good For The Game
Classic clown hijinks now enabled

## Changelog

:cl: Ryll
add: You can now prop things on top of airlocks to drop onto people
/:cl:

### CODE REVIEW

**Comment by Ryll-Ryll** [comment]
CODE THEIF CODE THIEF CODE THIEF

**Comment by saprasam** [comment]
i kneel

**Comment by ATH1909** [comment]
any mix that's too degenerate for a door-mounted bucket would also be too degenerate for a hand-applied bucket, a thrown bottle, or a beaker+beaker grenade. I don't really see the point in restricting the door-mounted buckets to water.

**Comment by ATH1909** [comment]
is this going to interfere with prying open doors with fire axes?

File: code/datums/components/airlock_tip.dm [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	SIGNAL_HANDLER
+
+	if(!istype(targeted_atom, /obj/machinery/door))
+		return
+
+	INVOKE_ASYNC(src, PROC_REF(try_plant), our_item, targeted_atom, user)
+	return TRUE
+	//if(targeted_door.GetComponent(/datum/component/)) // check for if something's already on it
+	//if(!check_can_plant(targeted_atom, user))
+		//return
```
Comment (MrMelbert) [inline code/datums/components/airlock_tip.dm:50]: finish your code

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/game/objects/items/fireaxe.dm [hunk 2]
```diff
@@ -40,6 +40,10 @@
 		bonus_modifier = 0 , \
 		butcher_sound = hitsound, \
 	)
+	var/datum/callback/door_start_trap_callback = CALLBACK(src, PROC_REF(start_trapping_door))
+	var/datum/callback/airlock_tip_callback = CALLBACK(src, PROC_REF(on_airlock_tip))
+
+	AddComponent(/datum/component/airlock_tip, 1 MINUTES, door_start_trap_callback, airlock_tip_callback)
```
Comment (MrMelbert) [inline code/game/objects/items/fireaxe.dm:46]: also this prevents bashing doors down with a fireaxe

**Comment by Ryll-Ryll** [comment]
> is this going to interfere with prying open doors with fire axes?

probably, i sorta added fireaxes as a proof of concept for the component

File: code/game/objects/items/fireaxe.dm (hunk 2 above)
Comment (Wallemations) [reply code/game/objects/items/fireaxe.dm:46]: Would this also interfere with the crowbarring aspect of fire axes?

**Comment by Wallemations** [comment]
While I love the humor aspect of this, I'm struggling to figure out how this works from a special perspective. How are you managing to hang this on top of an _airlock_? It's not a swinging door you can discreetly prop open and put something on top, it's a sliding airtight mechanism with (presumably) a roof above it blocking anything from getting on there.

Are you sticking an iron rod in the door as it closes and hanging the bucket off that?
All of these questions could probably be helped by showing some screenshots honestly

**Comment by moocowswag** [comment]
Im assuming its portrayed as literally just ontop of the airlock and when both the left and right parts of it seperate and make a gap it falls through

**Comment by optimumtact** [comment]
> While I love the humor aspect of this, I'm struggling to figure out how this works from a special perspective. How are you managing to hang this on top of an _airlock_? It's not a swinging door you can discreetly prop open and put something on top, it's a sliding airtight mechanism with (presumably) a roof above it blocking anything from getting on there.
> 
> Are you sticking an iron rod in the door as it closes and hanging the bucket off that? All of these questions could probably be helped by showing some screenshots honestly

airlocks have special notches for buckets

**Pull request closed without merging**